smart_open/smart_open_lib.py

```diff
@@ -339,6 +339,15 @@ def __init__(self, key):
         self.stream = _S3ReadStream(key)
         super(S3ReadStream, self).__init__(self.stream)
 
+        #
+        # Generator behavior changed in Python 3.
```

Comment:
@piskvorky Check out the next commit, it's possible to avoid the whole generator business entirely.